addons/sale_timesheet/controllers/portal.py

```diff
@@ -64,6 +64,12 @@ def _get_groupby_mapping(self):
             invoice='timesheet_invoice_id')
         return groupby_mapping
 
+    def _get_searchbar_sortings(self):
+        searchbar_sortings = super()._get_searchbar_sortings()
+        searchbar_sortings.update(
+            sol={'label': _('Sales Order Item'), 'order': 'timesheet_invoice_id'})
```

Comment:
The sales order line field in timesheet is `so_line` and not `timesheet_invoice_id`. 🙂 
```suggestion
            sol={'label': _('Sales Order Item'), 'order': 'so_line'})
```

Reply:
@xavierbol Done. Thanks.